src/ui/units/wizard/components/ValuesList/ValuesList.tsx

```diff
@@ -388,6 +393,22 @@ class ValuesList extends React.Component<Props, State> {
 
         this.props.onChangeSelectedValue(values[0] || null);
     };
+
+    private buildMultipleColorsDistincts(distincts: string[][], index: number): string[] {
+        const root = distincts[index];
+        if (distincts.length - 1 === index) {
+            return root;
+        }
+
+        return root.reduce((acc, v) => {
+            return [
+                ...acc,
+                ...this.buildMultipleColorsDistincts(distincts, index + 1).map(
+                    (v2) => `${v}; ${v2}`,
+                ),
+            ];
+        }, [] as string[]);
+    }
```

Comment:
I think It is necessary to put the algorithm for determining the keys for the dictionary in one place(somewhere in shared) - so that the client and server use the same code

https://github.com/datalens-tech/datalens-ui/blob/e0d63821b25243d9f72fc089c5ffa2265b7bc4e4/src/server/modes/charts/plugins/datalens/preparers/line-time/index.ts#L77

Reply:
I have extracted the very method of generating legend values. 

Generating all legend elements is a completely different approach on the server and client side. So I believe this should be changed in a different PR and for all preparations.